File: sdks/ruby/lib/novabilling/api_keys/types/create_api_key_body_dto.rb
# frozen_string_literal: true

module Novabilling
  module APIKeys
    module Types
      class CreateAPIKeyBodyDto < Internal::Types::Model
        field :name, -> { String }, optional: false, nullable: false
        field :scopes, -> { Internal::Types::Array[String] }, optional: false, nullable: false
        field :expires_at, -> { String }, optional: true, nullable: false, api_name: "expiresAt"
      end
    end
  end
end

File: sdks/php/src/ApiKeys/Requests/CreateApiKeyBodyDto.php
<?php

namespace NovaBilling\ApiKeys\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;
use NovaBilling\Core\Types\ArrayType;

class CreateApiKeyBodyDto extends JsonSerializableType
{
    /**
     * @var string $name
     */
    #[JsonProperty('name')]
    public string $name;

    /**
     * @var array<string> $scopes
     */
    #[JsonProperty('scopes'), ArrayType(['string'])]
    public array $scopes;

    /**
     * @var ?string $expiresAt
     */
    #[JsonProperty('expiresAt')]
    public ?string $expiresAt;

    /**
     * @param array{
     *   name: string,
     *   scopes: array<string>,
     *   expiresAt?: ?string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->name = $values['name'];
        $this->scopes = $values['scopes'];
        $this->expiresAt = $values['expiresAt'] ?? null;
    }
}
